project/source/Scripts/Objects/LabObject.gd

```diff
@@ -127,44 +131,45 @@ func StartDragging():
 	dragStartGlobalPos = global_position
 
 #sometimes an object needs to stop dragging without interacting (eg. when an object that can't change subscenes is dragging, and the mouse leaves its subscene)
-func StopDragging(action: bool = true):
+func StopDragging(action: bool = true) -> void:
 	dragging = false
 	set_linear_velocity(Vector2(0, 0))
-	mode = defaultMode
+	#mode = defaultMode
+	set_freeze_mode(RigidBody2D.FREEZE_MODE_STATIC)
 	z_index = defaultZIndex
 	z_as_relative = defaultZAsRelative
 	if action: OnUserAction()
 
-func DragMove():
+func DragMove() -> void:
 	global_position = get_global_mouse_position() - dragOffset
 
 func ClampObjectPosition() -> void:
-	var currentModule = GetCurrentModuleScene()
-	var labBoundary = ""
+	var currentModule:Node = GetCurrentModuleScene()
+	var labBoundary:Node = null
 	for child in currentModule.get_children():
 		if child.name == "LabBoundary":
 			labBoundary = child
-	if typeof(labBoundary) != TYPE_STRING:
+	if labBoundary != null:
 		global_position.x = clamp(global_position.x, labBoundary.xBounds[0], labBoundary.xBounds[1])
 		global_position.y = clamp(global_position.y, labBoundary.yBounds[0], labBoundary.yBounds[1])
 
-func GetIntersectingLabObjects():
-	var spaceState = get_world_2d().direct_space_state
-	var queryResults = []
+func GetIntersectingLabObjects() -> Array[LabObject]:
+	var spaceState := get_world_2d().direct_space_state
+	var queryResults: Array[LabObject] = []
```

Comment:
Change
```var queryResults: Array[LabObject] = []```

to

```var queryResults: Array[Dictionary] = []```

Reply:
Yeah I just noticed that while trying to figure out why it wasn't working. It seems to just... silently ignore when you try to `append_array` with the wrong type?